src/app/components/skill-list/skill-list.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, OnInit } from '@angular/core';
+import { Skill } from 'src/app/models/skill';
+import { SkillService } from 'src/app/services/skill-service.service'
+import { ActivatedRoute, Router } from '@angular/router';
+ 
+@Component({
+  selector: 'app-skill-list',
+  templateUrl: './skill-list.component.html',
+  styleUrls: ['./skill-list.component.css']
+})
+export class SkillListComponent implements OnInit {
+ 
+  skills: Skill[];
+ 
+  deleteRow(skill: Skill) {
+    console.log("delete this row:" + skill.name);
+    this.skillService.deleteSkill(skill).subscribe(() => this.gotoSkillListAfterDelete());
+  }
+
+  relinkSkills() {
+    console.log("relink skills");
+    this.skillService.relinkSkills();
+  }
+
+  navigateVacancies() {
+    this.router.navigate(['']);
+  }
+
+  addSkill() {
+    this.router.navigate(['addskill']);
+  }
+
+  constructor(
```

Comment:
For readability of the code it might be an idea to move the constructor above your methods. After your constructor it can be useful to first have the `ngOnInit` as that's the function that's called upon initialization.

Reply:
moved